lib/data_processing.py

```diff
@@ -61,7 +61,21 @@ def load_data(input_file='input_delivery.csv.gz', input_path="output"):
         covid_vacc_flag_mod = np.where(df["covid_vacc_moderna_date"]!=0, 1, 0),
         covid_vacc_2nd = np.where(df["covid_vacc_second_dose_date"]!=0, 1, 0),
         covid_vacc_bin = np.where(df["covid_vacc_date"]!=0, 1, 0))
-
+        
+    # Mixed doses:
+    # flag patients with a second dose, where more than one type has been recorded
+    df = df.assign(    
+        covid_vacc_ox_pfz = np.where((df["covid_vacc_2nd"]==1) & \
+                                          (df["covid_vacc_flag_pfz"]==1) & \
+                                          (df["covid_vacc_flag_ox"]==1), 1, 0),
+        covid_vacc_ox_mod = np.where((df["covid_vacc_2nd"]==1) & \
+                                          (df["covid_vacc_flag_ox"]==1) & \
+                                          (df["covid_vacc_flag_mod"]==1), 1, 0),
+        covid_vacc_mod_pfz = np.where((df["covid_vacc_2nd"]==1) & \
+                                          (df["covid_vacc_flag_mod"]==1) & \
+                                          (df["covid_vacc_flag_pfz"]==1), 1, 0),
```

Comment:
Here we check for `covid_vacc_2nd` being present because we have already stipulated that this vaccination record must have been at least 2/3 weeks after the first recorded dose. This ensures that we don't count patients as double-vaccinated who had 2 doses recorded in close proximity, which are likely duplicate records rather than two separate doses. 

However, as ever, there will still be some other small data quality issues, e.g. missingness, but these issues should only affect a very small minority of patients.